### PR TITLE
Add Tengam materials

### PR DESCRIPTION
**Pull request by glowredman**
Rough idea by @minecraft7771. Raw Tengam will be a T9 ore. Attuned Tengam will replace Magnetic Samarium in UEV+ recipes.

### CODE REVIEW

File: src/main/java/gregtech/loaders/postload/recipes/ElectromagneticSeparatorRecipes.java [hunk 1]
```diff
@@ -22,5 +22,17 @@ public void run() {
...
+            .itemInputs(Materials.TengamRaw.getDust(1))
+            .itemOutputs(
+                Materials.TengamPurified.getDust(1),
+                Materials.NeodymiumMagnetic.getDust(1),
+                Materials.SamariumMagnetic.getDust(1))
+            .outputChances(10000, 1000, 1000)
+            .noFluidInputs()
+            .noFluidOutputs()
+            .duration(10 * SECONDS)
+            .eut(TierEU.RECIPE_UHV / 2)
```
Comment (minecraft7771) [inline src/main/java/gregtech/loaders/postload/recipes/ElectromagneticSeparatorRecipes.java:35]: We had problems with half tier voltages in the past. Would be better to stick to a full amp imo.

File: src/main/java/gregtech/loaders/oreprocessing/ProcessingTransforming.java [hunk 2]
```diff
@@ -162,6 +162,17 @@ public void registerOre(OrePrefixes aPrefix, Materials aMaterial, String aOreDic
...
                 break;
 
+            case "TengamPurified":
+            // Polarizer recipes
+            {
+                GT_Values.RA.addPolarizerRecipe(
+                    GT_Utility.copyAmount(1L, aStack),
+                    GT_OreDictUnificator.get(aPrefix, Materials.TengamAttuned, 1L),
+                    (int) Math.max(16L, aPrefix.mMaterialAmount * 128L / GT_Values.M),
+                    (int) TierEU.UHV / 2);
```
Comment (minecraft7771) [inline src/main/java/gregtech/loaders/oreprocessing/ProcessingTransforming.java:172]: Apart from the point addressed below, why not Recipe_UHV?

**Review by minecraft7771 — COMMENTED** [review]
Just a style review. As for balance this is better reviewed by @Connor-Colenso

File: src/main/java/gregtech/loaders/postload/recipes/ElectromagneticSeparatorRecipes.java (hunk 1 above)
Comment (glowredman) [reply src/main/java/gregtech/loaders/postload/recipes/ElectromagneticSeparatorRecipes.java:35]: I'm indifferent, I simply did it as the other recipe in that class.

File: src/main/java/gregtech/loaders/oreprocessing/ProcessingTransforming.java (hunk 2 above)
Comment (glowredman) [reply src/main/java/gregtech/loaders/oreprocessing/ProcessingTransforming.java:172]: I kept it the same as the other Polarizer recipes.

**Review by minecraft7771 — APPROVED** [review]

**Pull request merged**